### DISCUSSION TITLE
I don't know how to use this

### DISCUSSION THREAD

**Discussion by AntonioDLC** [argrecsys/arg-miner#87] [Q&A]
Hi, I'm really interested in your argument miner but I really can't figure out where to even begin
I see the java classes deep into the code folder but see no main method, and I can't see any code or console command sample to use it to parse personalized documents and get the results just like you did.

**Comment by ansegura7** [comment]
Hi @AntonioDLC

Thank you for your interest in this project 😃 

The **Argument Miner** codebase is located in the folder [/code](https://github.com/argrecsys/arg-miner/tree/main/code). And specifically, the _main_ function of the project is the [RecSys.java](https://github.com/argrecsys/arg-miner/blob/main/code/ArgumentMiner/src/es/uam/irg/recsys/RecSys.java) file, which reads certain configuration parameters (which can be modifiable) and executes the miner.

Feel free to explore this project and the associated papers, and do not hesitate to write to us if you have any questions.

**Reply by AntonioDLC** [reply]
Ok I see that thanks!
Can you tell me what command do you use to build and from which directory?
I'm not too knowledgeable of java and I've been trying javac commands for a while, trying to figure out the way to address the library jars and src folder for the package importing to work

**Reply by AntonioDLC** [reply]
Did you use Maven or Gradle? If so, do you have the project .xml or .gradle with the list of dependencies?
The java files use the edu.stanford.nlp library but it's not provided in the lib/ folder.

**Reply by AntonioDLC** [reply]
I've downloaded it from here https://repo1.maven.org/maven2/edu/stanford/nlp/stanford-corenlp/4.5.7/, copied the jar to /lib, and got it to build with the following commands while at the /code directory:
`find src -name "*.java" > sources.txt`
`javac -cp "lib/*" -d . @sources.txt`

**Reply by AntonioDLC** [reply]
And finally I managed to run it with: `java -cp ".:lib/*" es.uam.irg.recsys.RecSys`

**Reply by AntonioDLC** [reply]
Got an error about the database configs being all asterisks:
> Exception in thread "main" while scanning an alias
>  in "<reader>", line 1, column 12:
>     db_server: ****
>                ^
> expected alphabetic or numeric character, but found but found *
>  in "<reader>", line 1, column 13:
>     db_server: ****

**Reply by ansegura7** [reply]
Hi Antonio,

Thanks for your interest in ArgMiner.

The error you see is because this tool uses MongoDB to store the results of the argument mining, for which you must first configure the connection string to a MongoDB instance.

The file you must configure (and which currently has all fields with asterisks) is: https://github.com/argrecsys/arg-miner/blob/main/analysis/config/mdb_setup.yaml

You can download MongoDB Community from this link: https://www.mongodb.com/try/download/community

On the other hand, we would love to know more about what you want to use this tool for, so we would be very grateful if you could tell us, and we will know better how to help you.

**Reply by AntonioDLC** [reply]
Hi Andrés, what I want is to be able to mine for arguments and make argument graphs out of scientific papers (mainly).
In this moment of my life in particular I want to generate an argument graph for this document: https://nap.nationalacademies.org/catalog/10490/dietary-reference-intakes-for-energy-carbohydrate-fiber-fat-fatty-acids-cholesterol-protein-and-amino-acids
Having as "major claims" the amount of macronutrients we need to eat daily on average.

I made a partial graph, manually, about the carbohydrate need: https://antoniodlc.github.io/alimenergia_plus_docuparts_horizontal.svg
But making graphs manually at sentence level is a long and tedious task and since the natural language processing fields are strong with chat gpt advances and whatnot, I'm on the quest to find argument miners which can do the task automatically.

(Btw sé que ambos hablamos español pero capaz querés que esta discusión se mantenga en inglés por si lo chusmean otras personas. Avisame si seguimos en inglés.)

**Reply by ansegura7** [reply]
What you’ve shared is very interesting and certainly an excellent use case for automated argument extraction.

Would you be open to scheduling a meeting to discuss this further and explore how I can better support your research? If that works for you, feel free to email me with your availability (with time zone) for this or next week.
